src/simple_treeview.rs

```diff
@@ -0,0 +1,101 @@
+//! # TreeView Sample
+//!
+//! This sample demonstrates how to create a TreeView with either a ListStore or TreeStore.
+
+extern crate glib;
+extern crate gtk;
+
+use gtk::prelude::*;
+use gtk::{
+    CellRendererText, Label, ListStore, TreeView, TreeViewColumn,
+    Window, WindowPosition, WindowType
+};
+
+fn append_column(tree: &TreeView, id: i32) {
+    let column = TreeViewColumn::new();
+    let cell = CellRendererText::new();
+
+    column.pack_start(&cell, true);
+    // We associate the view's column with the model's `id` column.
+    column.add_attribute(&cell, "text", id);
+    tree.append_column(&column);
+}
+
+fn create_and_fill_model() -> ListStore {
+    // We create a model with two rows.
+    let store = ListStore::new(&[String::static_type(), u32::static_type()]);
+
+    // We fill the treeview.
+    let entries = &["Michel", "Sara", "Liam", "Zelda", "Neo", "Octopus master"];
+    for (i, entry) in entries.iter().enumerate() {
+        store.insert_with_values(None, &[0, 1], &[&entry, &(i as u32)]);
+    }
+    store
+}
+
+fn create_and_setup_view() -> TreeView {
+    // we create the tree view
+    let tree = TreeView::new();
+
+    tree.set_headers_visible(false);
+    // we create the two columns inside the view
+    append_column(&tree, 0);
+    append_column(&tree, 1);
+    tree
+}
+
+fn main() {
+    if gtk::init().is_err() {
+        println!("Failed to initialize GTK.");
+        return;
+    }
+
+    let window = Window::new(WindowType::Toplevel);
+
+    window.set_title("Simple TreeView example");
+    window.set_position(WindowPosition::Center);
+
+    window.connect_delete_event(|_, _| {
+        gtk::main_quit();
+        Inhibit(false)
+    });
+
+    // We create a vertical layout to place both tree view and label in the window.
+    let vertical_layout = gtk::Box::new(gtk::Orientation::Vertical, 0);
+
+    // We create the label, but because of the closure, we need to wrap it inside
+    // a Rc<Cell<>> to be able to use it outside the closure.
+    let label = Label::new(None);
+    let label_copy = label.clone();
```

Comment:
It's the "easy" way to say "internally, we're using g_ref to increment the number of references pointing to this object to avoid it going to the void when the destructor is called". :)